aws/README.md

```diff
@@ -1,3 +1,93 @@
+# Contributor process
+
+If you have a pull request that contains tests that need to be enabled in CI you can get the ball rolling by
+submitting a pull request. Start by updating the appropriate test policy in the policy directory (policy
+groups are defined below).
+
+Good integration tests should be contained in a block with an always statement that cleans up any stray resources
+if the integration tests fail. Despite this, bugs in the module and flaky AWS services could cause there to be leaked
+resources. Because of this, if your tests create a resource and the tests are run in CI, that resource must also have
+a terminator class. We deploy a lambda function that runs these terminator classes.
+
+To create your terminator class, you'll need to decide whether to use the Terminator base class or the DbTerminator
+base class.
+
+We terminate resources found based on their age. Not all AWS resources return the creation date timestamp so those
+resources get stored in a database with the time the terminator classes first found them and we approximate when to
+delete them from that. If the resource has a creation date timestamp use the Terminator base class. Otherwise use
+DbTerminator.
+
+Your terminator class should have the staticmethod `create`, the properties `id` and `name` (and `created_time` if
+you are using the Terminator base class), and the method `terminate`.
+
+The create method returns the credentials to create the client, the class name, the boto3 resource name to create the
+client, and a function for the client to use which will list all the given resources for that resource type.
+
+Here's an example:
+
+```python
+class Ec2Instance(Terminator):
+
+    @staticmethod
+    def create(credentials):
+
+        def get_instances(client):
+            return [i for r in client.describe_instances()['Reservations'] for i in r['Instances']]
+
+        return Terminator._create(credentials, Ec2Instance, 'ec2', get_instances)
+```
+
+The `id`, `name`, and `created_time` properties should return an appropriate value from self.instance which is
```

Comment:
Is it acceptable for `name` to not be unique? (e.g. two ACM certificates with the same domain, two Route53 hosted zones with the same domain, etc)

Reply:
Looking at existing code, it seems the answer is that they don't have to be unique. So maybe state that explicitly here?